ignite/engine/engine.py

```diff
@@ -139,9 +139,12 @@ def __init__(self, process_function: Callable[["Engine", Any], Any]):
         self.logger = logging.getLogger(__name__ + "." + self.__class__.__name__)
         self._process_function = process_function
         self.last_event_name: Optional[Events] = None
-        self.should_terminate = False
-        self.skip_completed_after_termination = False
-        self.should_terminate_single_epoch = False
+        # should_terminate flag: False - don't terminate, True - terminate,
+        # "skip_completed" - terminate and skip the event "COMPLETED"
+        self.should_terminate: bool | str = False
```

Comment:
I left this comment on the commit, but it is not shown in the PR, so duplicating it, sorry.


Concerning `|` operator for typing, we may need to check whether this is supported let's say since python 3.9 and 3.8, we still however have python 3.8 in our CI: https://github.com/pytorch/ignite/blob/master/.github/workflows/pytorch-version-tests.yml
For those users who still on python 3.8 I would still use `Union` instead, such that it wont break them even if they are using unsupported python version like 3.8 at least until the next year.

Reply:
It makes a lot of sense. I think it the pipe operator should be supported from 3.10 on (PEP 604)